docker/init.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+declare -r USER=astraea
+declare -r IMAGE_NAME="$REPO:$VERSION"
+declare -r BUILD=${BUILD:-false}
+declare -r RUN=${RUN:-true}
+declare -r DOCKER_FOLDER=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
+
+
+function checkDocker() {
+  if [[ "$(which docker)" == "" ]]; then
+    echo "you have to install docker"
+    exit 2
+  fi
+}
+
+function checkNetwork() {
+  if [[ "$ADDRESS" == "127.0.0.1" || "$ADDRESS" == "127.0.1.1" ]]; then
+    echo "Either 127.0.0.1 or 127.0.1.1 can't be used in this script. Please check /etc/hosts"
+    exit 2
+  fi
+}
+
+function buildImageIfNeed() {
+  if [[ "$(docker images -q $IMAGE_NAME 2>/dev/null)" == "" ]]; then
+    local needToBuild="true"
+    if [[ "$BUILD" == "false" ]]; then
+      docker pull $IMAGE_NAME 2>/dev/null
+      if [[ "$?" == "0" ]]; then
+        needToBuild="false"
+      else
+        echo "Can't find $IMAGE_NAME from repo. Will build $IMAGE_NAME on the local"
+      fi
+    fi
+    if [[ "$needToBuild" == "true" ]]; then
+      generateDockerfile
+      docker build --no-cache -t "$IMAGE_NAME" -f "$DOCKERFILE" "$DOCKER_FOLDER"
```

Comment:
Is `DOCKERFILE` defined?

Reply:
Sorry,I didn't notice that this function has a function calls to the original script.

Reply:
@chia7712 I later tested and found that when calling `generateDockerfile` function, the function can be called normally, and the `DOCKERFILE` at this time has also been defined.